### DISCUSSION TITLE
.NET Core 7 browser.LoadUrlAsync is not working - OffScreen

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [cefsharp/CefSharp#4329] [General]
### Bug Report

Version 107.1.120

Tested both target x86 / x64

.NET Core 7
    
Windows 10

OffScreen

**debug.log is empty**

Here all settings

![image](https://user-images.githubusercontent.com/19240467/206051030-1ba5cfe8-4cdc-4f5d-8016-f0efbed651a1.png)

![image](https://user-images.githubusercontent.com/19240467/206051068-eb391e78-13bc-4586-b971-7384c3c3626c.png)

**What happens is nothing happens simply. Only prints hello world**

Here the entire code

```
using CefSharp;
using CefSharp.OffScreen;

namespace cefsharp2
{
    internal class Program
    {
        static CefSettings settings = new CefSettings()
        {
            //By default CefSharp will use an in-memory cache, you need to specify a Cache Folder to persist data
            CachePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "CefSharp\\Cache")
        };

        static ChromiumWebBrowser browser = new ChromiumWebBrowser();

        const string srBaseUrl = "https://www.pokemonpets.com/";

        static async Task Main(string[] args)
        {
            Console.WriteLine("Hello World!");
            var success = await Cef.InitializeAsync(settings, performDependencyCheck: true, browserProcessHandler: null);


            if (!success)
            {
                throw new Exception("Unable to initialize CEF, check the log file.");
            }

             
            var initialLoadResponse = await browser.LoadUrlAsync(srBaseUrl);

            if (!initialLoadResponse.Success)
            {
                throw new Exception(string.Format("Page load failed with ErrorCode:{0}, HttpStatusCode:{1}", initialLoadResponse.ErrorCode, initialLoadResponse.HttpStatusCode));
            }
            else
            {
                Console.WriteLine("print success");
            }
        }
    }
}
```

Here the output result

![image](https://user-images.githubusercontent.com/19240467/206051305-17dc1025-882f-4c7b-b6c6-5312c927582b.png)

**The output debug in visual studio**

```
'cefsharp2.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App\7.0.0\System.Diagnostics.Process.dll'. 
'cefsharp2.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App\7.0.0\System.ComponentModel.Primitives.dll'. 
'cefsharp2.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files (x86)\dotnet\shared\Microsoft.NETCore.App\7.0.0\System.Private.Uri.dll'. 
Exception thrown: 'System.Exception' in CefSharp.Core.Runtime.dll
The thread 0x9928 has exited with code 0 (0x0).
The thread 0xbdd4 has exited with code 0 (0x0).
The thread 0x5718 has exited with code 0 (0x0).
The thread 0xb064 has exited with code 0 (0x0).
```

the output folder

![image](https://user-images.githubusercontent.com/19240467/206053750-97743cb0-ec22-4e05-89b5-819b89fd166a.png)

**Comment by FurkanGozukara** [comment]
By the way I couldn't find and .net core example

The .net framework example works I have tested

**Reply by amaitland** [reply]
In future please use [Discussions](https://github.com/cefsharp/CefSharp/discussions) to ask support, just because it's not working doesn't mean it's a bug.

> By the way I couldn't find and .net core example

There are examples at https://github.com/cefsharp/CefSharp.MinimalExample

There are different solutions targeting different frameworks.

> in synchronized version code works

Async code requires a [SynchronizationContext](https://learn.microsoft.com/en-us/dotnet/api/system.threading.synchronizationcontext?view=net-7.0) to work correctly.

- Console applications by default don't have a sync context, you need to add your own,
- Your code is missing a call to Cef.Shutdown, cache won't be persisted properly without it
- Cef.Initialize(Async) must be called before you create the first `ChromiumWebBrowser` instance, in your case it's static so the ordering will likely be problematic.
- Pass the initial url into the `ChromiumWebBrowser` constructor and use `WaitForInitialLoadAsync` for first page load. Use `LoadUrlAsync` for subsiquent.

There are multiple async demos for the OffScreen implementation:
- https://github.com/cefsharp/CefSharp.MinimalExample/blob/master/CefSharp.MinimalExample.OffScreen/Program.cs#L48
- https://github.com/cefsharp/CefSharp/blob/master/CefSharp.OffScreen.Example/Program.cs#L33

**Comment by FurkanGozukara** [comment]
in synchronized version code works
